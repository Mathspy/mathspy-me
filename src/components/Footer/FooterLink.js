import React from "react";
import { Link } from "gatsby";

import { css } from "emotion";

const FooterLink = ({ text, to }) => (
  <Link className={linkStyle} to={to}>
    {text}
  </Link>
);

const linkStyle = css`
  color: white;
  text-decoration: none;

  &:hover {
    color: #e56a17;
  }
`;

export default FooterLink;
