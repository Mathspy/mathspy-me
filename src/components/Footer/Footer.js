import React from "react";

import { css } from "emotion";

import FooterLink from "./FooterLink";

const Footer = ({ links }) => (
  <div className={containerStyle}>
    <h2>Other recordings of my adventures:</h2>
    <div className={wrapperStyle}>
      {links.map(link => (
        <FooterLink key={link.id} to={link.url} text={link.text} />
      ))}
    </div>
  </div>
);

const containerStyle = css`
  background-color: #3d1755;
  padding: 7px;

  h2 {
    color: white;
    margin: 10px 0;
  }

  @media (min-width: 420px) {
    padding: 12px;
  }

  @media (min-width: 900px) {
    padding: 17px;
    flex: 3;
    max-width: 368px;
  }
`;

const wrapperStyle = css`
  padding-left: 20px;
  border-left: solid 1px #e56a17;
  a {
    display: block;
    margin-bottom: 8px;
  }
`;

export default Footer;
