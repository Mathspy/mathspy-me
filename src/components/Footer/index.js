import React from "react";
import { StaticQuery, graphql } from "gatsby";

import Footer from "./Footer";

export default ({ pathname }) => {
  return (
    <StaticQuery
      query={graphql`
        query {
          allMarkdownRemark(
            sort: { fields: [frontmatter___date], order: DESC }
            limit: 4
          ) {
            edges {
              node {
                id
                frontmatter {
                  title
                }
                fields {
                  slug
                }
              }
            }
          }
        }
      `}
      render={data => {
        const edges = data.allMarkdownRemark.edges
          .filter(edge => edge.node.fields.slug !== pathname)
          .slice(0, 3);
        return (
          <Footer
            links={edges.map(edge => ({
              id: edge.node.id,
              text: edge.node.frontmatter.title,
              url: edge.node.fields.slug,
            }))}
          />
        );
      }}
    />
  );
};
