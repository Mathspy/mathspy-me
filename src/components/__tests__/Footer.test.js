import React from "react";

import { shallow } from "enzyme";

import Footer from "../Footer/Footer";
import FooterLink from "../Footer/FooterLink";

describe("Footer", () => {
  it("should be a div containing a catchphrase and another div", () => {
    const wrapper = shallow(<Footer links={[]} />);

    expect(wrapper.is("div")).toBe(true);
    expect(
      wrapper
        .children()
        .first()
        .is("h2")
    ).toBe(true);
    expect(
      wrapper
        .children()
        .last()
        .is("div")
    ).toBe(true);
  });

  it("should render the links passed to it as FooterLinks", () => {
    const wrapper = shallow(<Footer links={links} />);

    expect(wrapper.find(FooterLink)).toHaveLength(4);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<Footer links={links} />);

    expect(wrapper).toMatchSnapshot();
  });
});

const links = [
  {
    id: "0",
    url: "/blog/123/",
    text: "This blog post loves YOU!",
  },
  {
    id: "1",
    url: "/notblog/123/",
    text: "I am bad at writing good stories!",
  },
  {
    id: "2",
    url: "/otherpage/",
    text:
      "Once upon a time I was making a website with Gatsby, Jest and Storybook, and it was a lot of fun",
  },
  {
    id: "3",
    url: "https://www.google.com",

    text: "How to search and all that",
  },
];
