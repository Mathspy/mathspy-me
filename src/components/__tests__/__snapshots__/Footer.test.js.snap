// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
.emotion-1 {
  background-color: #3d1755;
  padding: 7px;
}

.emotion-1 h2 {
  color: white;
  margin: 10px 0;
}

@media (min-width:420px) {
  .emotion-1 {
    padding: 12px;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    padding: 17px;
    -webkit-flex: 3;
    -ms-flex: 3;
    flex: 3;
    max-width: 368px;
  }
}

.emotion-0 {
  padding-left: 20px;
  border-left: solid 1px #e56a17;
}

.emotion-0 a {
  display: block;
  margin-bottom: 8px;
}

<div
  className="emotion-1"
>
  <h2>
    Other recordings of my adventures:
  </h2>
  <div
    className="emotion-0"
  >
    <FooterLink
      key="0"
      text="This blog post loves YOU!"
      to="/blog/123/"
    />
    <FooterLink
      key="1"
      text="I am bad at writing good stories!"
      to="/notblog/123/"
    />
    <FooterLink
      key="2"
      text="Once upon a time I was making a website with Gatsby, Jest and Storybook, and it was a lot of fun"
      to="/otherpage/"
    />
    <FooterLink
      key="3"
      text="How to search and all that"
      to="https://www.google.com"
    />
  </div>
</div>
`;
